From 73bd937d975076c3f7754919d49d2610401dcf1b Mon Sep 17 00:00:00 2001
From: Mauro Carvalho Chehab <mchehab+samsung@kernel.org>
Date: Tue, 18 Jun 2019 16:06:08 -0300
Subject: [PATCH 34/55] docs: perf: move to the admin-guide

mainline inclusion
from mainline-v5.3-rc1
commit 59809fe88224db24432ad50e62fd8d5f0df738a1
bugzilla: https://gitee.com/openeuler/kernel/issues/I8AU2M

Reference: https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=59809fe88224db24432ad50e62fd8d5f0df738a1

-------------------------------------------------------------

The perf infrastructure is used for userspace to track issues.
At least a good part of what's described here is related to
it.

So, add it to the admin-guide.

Signed-off-by: Mauro Carvalho Chehab <mchehab+samsung@kernel.org>
Signed-off-by: hongrongxuan <hongrongxuan@huawei.com>
---
 Documentation/admin-guide/index.rst                  | 1 +
 Documentation/{ => admin-guide}/perf/arm-ccn.rst     | 0
 Documentation/{ => admin-guide}/perf/arm_dsu_pmu.rst | 0
 Documentation/{ => admin-guide}/perf/hisi-pmu.rst    | 0
 Documentation/{ => admin-guide}/perf/index.rst       | 2 --
 Documentation/{ => admin-guide}/perf/qcom_l2_pmu.rst | 0
 Documentation/{ => admin-guide}/perf/qcom_l3_pmu.rst | 0
 Documentation/{ => admin-guide}/perf/xgene-pmu.rst   | 0
 MAINTAINERS                                          | 4 ++--
 drivers/perf/qcom_l3_pmu.c                           | 2 +-
 10 files changed, 4 insertions(+), 5 deletions(-)
 rename Documentation/{ => admin-guide}/perf/arm-ccn.rst (100%)
 rename Documentation/{ => admin-guide}/perf/arm_dsu_pmu.rst (100%)
 rename Documentation/{ => admin-guide}/perf/hisi-pmu.rst (100%)
 rename Documentation/{ => admin-guide}/perf/index.rst (95%)
 rename Documentation/{ => admin-guide}/perf/qcom_l2_pmu.rst (100%)
 rename Documentation/{ => admin-guide}/perf/qcom_l3_pmu.rst (100%)
 rename Documentation/{ => admin-guide}/perf/xgene-pmu.rst (100%)

diff --git a/Documentation/admin-guide/index.rst b/Documentation/admin-guide/index.rst
index 89abc5057349..b75b6ae6f662 100644
--- a/Documentation/admin-guide/index.rst
+++ b/Documentation/admin-guide/index.rst
@@ -38,6 +38,7 @@ problems and bugs in particular.
    ramoops
    dynamic-debug-howto
    init
+   perf/index
 
 This is the beginning of a section with information of interest to
 application developers.  Documents covering various aspects of the kernel
diff --git a/Documentation/perf/arm-ccn.rst b/Documentation/admin-guide/perf/arm-ccn.rst
similarity index 100%
rename from Documentation/perf/arm-ccn.rst
rename to Documentation/admin-guide/perf/arm-ccn.rst
diff --git a/Documentation/perf/arm_dsu_pmu.rst b/Documentation/admin-guide/perf/arm_dsu_pmu.rst
similarity index 100%
rename from Documentation/perf/arm_dsu_pmu.rst
rename to Documentation/admin-guide/perf/arm_dsu_pmu.rst
diff --git a/Documentation/perf/hisi-pmu.rst b/Documentation/admin-guide/perf/hisi-pmu.rst
similarity index 100%
rename from Documentation/perf/hisi-pmu.rst
rename to Documentation/admin-guide/perf/hisi-pmu.rst
diff --git a/Documentation/perf/index.rst b/Documentation/admin-guide/perf/index.rst
similarity index 95%
rename from Documentation/perf/index.rst
rename to Documentation/admin-guide/perf/index.rst
index 4bf848e27f26..9d445451ea18 100644
--- a/Documentation/perf/index.rst
+++ b/Documentation/admin-guide/perf/index.rst
@@ -1,5 +1,3 @@
-:orphan:
-
 ===========================
 Performance monitor support
 ===========================
diff --git a/Documentation/perf/qcom_l2_pmu.rst b/Documentation/admin-guide/perf/qcom_l2_pmu.rst
similarity index 100%
rename from Documentation/perf/qcom_l2_pmu.rst
rename to Documentation/admin-guide/perf/qcom_l2_pmu.rst
diff --git a/Documentation/perf/qcom_l3_pmu.rst b/Documentation/admin-guide/perf/qcom_l3_pmu.rst
similarity index 100%
rename from Documentation/perf/qcom_l3_pmu.rst
rename to Documentation/admin-guide/perf/qcom_l3_pmu.rst
diff --git a/Documentation/perf/xgene-pmu.rst b/Documentation/admin-guide/perf/xgene-pmu.rst
similarity index 100%
rename from Documentation/perf/xgene-pmu.rst
rename to Documentation/admin-guide/perf/xgene-pmu.rst
diff --git a/MAINTAINERS b/MAINTAINERS
index 600f432e2e16..e9302e09600b 100644
--- a/MAINTAINERS
+++ b/MAINTAINERS
@@ -1040,7 +1040,7 @@ APPLIED MICRO (APM) X-GENE SOC PMU
 M:	Tai Nguyen <ttnguyen@apm.com>
 S:	Supported
 F:	drivers/perf/xgene_pmu.c
-F:	Documentation/perf/xgene-pmu.rst
+F:	Documentation/admin-guide/perf/xgene-pmu.rst
 F:	Documentation/devicetree/bindings/perf/apm-xgene-pmu.txt
 
 APTINA CAMERA SENSOR PLL
@@ -6666,7 +6666,7 @@ W:	http://www.hisilicon.com
 F:	Documentation/admin-guide/perf/hisi-pcie-pmu.rst
 F:	Documentation/admin-guide/perf/hisi-pmu.rst
 F:	drivers/perf/hisilicon
-F:	Documentation/perf/hisi-pmu.rst
+F:	Documentation/admin-guide/perf/hisi-pmu.rst
 
 HISILICON PTT DRIVER
 M:	Yicong Yang <yangyicong@hisilicon.com>
diff --git a/drivers/perf/qcom_l3_pmu.c b/drivers/perf/qcom_l3_pmu.c
index cc88dc92577e..6cefee509e57 100644
--- a/drivers/perf/qcom_l3_pmu.c
+++ b/drivers/perf/qcom_l3_pmu.c
@@ -7,7 +7,7 @@
  * the slices. User space needs to aggregate to individual counts to provide
  * a global picture.
  *
- * See Documentation/perf/qcom_l3_pmu.rst for more details.
+ * See Documentation/admin-guide/perf/qcom_l3_pmu.rst for more details.
  *
  * Copyright (c) 2015-2017, The Linux Foundation. All rights reserved.
  *
-- 
2.27.0

